### PR TITLE
Updates unconfirmed funds modal

### PR DESCRIPTION
**Pull request by AdimekweEbuka**
PR to solve #707 
godcr (testnet) 2021-12-12 20-07-49

### CODE REVIEW

**Review by devchoplife — CHANGES_REQUESTED** [review]
The size of the info icon should be increased a bit. 
Screenshot from 2021-12-09 11-18-27

File: ui/modal/info_modal.go [hunk 1]
```diff
@@ -73,12 +81,23 @@ func (in *InfoModal) Dismiss() {
...
+func (in *InfoModal) ShowTextEditor(showTextEditor bool) *InfoModal {
+	in.showEditor = showTextEditor
+	return in
+}
+
+func (in *InfoModal) TextInputEditor(editor decredmaterial.Editor) *InfoModal {
+	in.textInputEditor = editor
+	return in
+}
+
```
Comment (Sirmorrison) [inline ui/modal/info_modal.go]: Kindly undo the changes to this modal.

We have the text input modal for this already.

File: ui/modal/info_modal.go [hunk 2]
```diff
@@ -215,6 +255,19 @@ func (in *InfoModal) Layout(gtx layout.Context) D {
...
+			return layout.Dimensions{}
+		}
+
+		return layout.Inset{Top: values.MarginPaddingMinus5, Left: values.MarginPaddingMinus5}.Layout(gtx, func(gtx C) D {
+			in.textInputEditor.TextSize = values.TextSize14
+			in.textInputEditor.EditorStyle.Color = in.Theme.Color.Black
+			in.textInputEditor.EditorStyle.Font.Weight = text.Weight(in.Theme.TextSize.U)
+			return in.textInputEditor.Layout(gtx)
+		})
+	}
```
Comment (Sirmorrison) [inline ui/modal/info_modal.go]: revert this please.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

File: ui/modal/info_modal.go [hunk 3]
```diff
@@ -167,14 +170,16 @@ func (in *InfoModal) handleEnterKeypress() {
 }
 
 func (in *InfoModal) Handle() {
+
+
```
Comment (devchoplife) [inline ui/modal/info_modal.go]: Remove whitespaces

File: ui/modal/info_modal.go [hunk 4]
```diff
@@ -167,14 +170,16 @@ func (in *InfoModal) handleEnterKeypress() {
...
 	for in.btnPositve.Clicked() {
-		in.DismissModal(in)
-		in.positiveButtonClicked()
+			in.DismissModal(in)
+			in.positiveButtonClicked()
```
Comment (devchoplife) [inline ui/modal/info_modal.go]: Correct indentation

File: ui/modal/info_modal.go [hunk 5]
```diff
@@ -167,14 +170,16 @@ func (in *InfoModal) handleEnterKeypress() {
...
 	}
 
 	for in.btnNegative.Clicked() {
-		in.DismissModal(in)
-		in.negativeButtonClicked()
+			in.DismissModal(in)
+			in.negativeButtonClicked()
```
Comment (devchoplife) [inline ui/modal/info_modal.go]: Correct indentation here also

File: ui/modal/info_modal_layouts.go [hunk 6]
```diff
@@ -121,3 +123,30 @@ func backupInfo(th *decredmaterial.Theme) []layout.Widget {
...
+
+		func(gtx C) D {
+			text := `<span style="text-color: gray">
+					Please type "<span style="text-color: black"> <span style="font-weight: bold">I understand the risks</span> </span>
+					" to allow spending from unmixed accounts.
+			</span>`
+			return renderers.RenderHTML(text, th).Layout(gtx)
+		},
+	}
+}
```
Comment (devchoplife) [inline ui/modal/info_modal_layouts.go]: Add newline to end of file

File: ui/modal/text_input_modal.go [hunk 7]
```diff
@@ -87,9 +90,14 @@ func (tm *TextInputModal) SetCancelable(min bool) *TextInputModal {
 	return tm
 }
 
+func(tm *TextInputModal) SetTextWithTemplate() *TextInputModal {
+tm.textCustomTemplate = allowUnspendUnmixedAcct(tm.Theme)
+return tm
```
Comment (devchoplife) [inline ui/modal/text_input_modal.go]: Correct indentation here

File: ui/page/wallets/privacy_page.go [hunk 8]
```diff
@@ -37,6 +37,8 @@ type PrivacyPage struct {
 	toggleMixer             *decredmaterial.Switch
 	allowUnspendUnmixedAcct *decredmaterial.Switch
 
+	
+
```
Comment (devchoplife) [inline ui/page/wallets/privacy_page.go]: Remove whitespaces

File: ui/page/wallets/privacy_page.go [hunk 9]
```diff
@@ -295,26 +298,37 @@ func (pg *PrivacyPage) Handle() {
...
+				} else {
 					pg.wallet.SetBoolConfigValueForKey(dcrlibwallet.AccountMixerConfigSet, false)
-				}).
-				Show()
+					tim.Dismiss()
+				}
+				return tim.IsLoading
+			})
+
+			textModal.Title("Confirm to allow spending from unmixed accounts").
```
Comment (devchoplife) [inline ui/page/wallets/privacy_page.go:314]: Create title stringin values package and call within here
Here after this line:
https://github.com/planetdecred/godcr/blob/796ab852f937b39b90f579447099616cca7dd58b/ui/values/strings.go#L230

File: ui/modal/info_modal.go [hunk 10]
```diff
@@ -147,6 +147,9 @@ func (in *InfoModal) SetupWithTemplate(template string) *InfoModal {
 		customTemplate = setupMixerInfo(in.Theme)
 	case WalletBackupInfoTemplate:
 		customTemplate = backupInfo(in.Theme)
+	case AllowUnmixedSpendingTemplate:
+		title = "Confirm to allow spending from unmixed accounts"
+		customTemplate = allowUnspendUnmixedAcct(in.Theme)
```
Comment (devchoplife) [inline ui/modal/info_modal.go]: This is redundant
Since the layout of the text in the text input modal is handled by a function, it does not need an info modal template

**Review by devchoplife — CHANGES_REQUESTED** [review]

**Comment by devchoplife** [comment]
Please add an image to the PR description and rebase as PR has conflicts
@AdimekweEbuka

File: ui/page/wallets/privacy_page.go (hunk 9 above)
Comment (AdimekweEbuka) [reply ui/page/wallets/privacy_page.go:314]: Since the title is only being used once, I didn't think it needed to be declared in the strings package

Comment (devchoplife) [reply ui/page/wallets/privacy_page.go:314]: Yeah but most of the strings there are specific to different models and are used just once. It is just a way to keep things organized

Comment (AdimekweEbuka) [reply ui/page/wallets/privacy_page.go:314]: Alright, noted. Thanks.

Comment (Sirmorrison) [reply ui/page/wallets/privacy_page.go:314]: The values /strings.go file should not be modified at any point please. Kindly hard code your string variables.

File: ui/page/wallets/privacy_page.go [hunk 11]
```diff
@@ -295,25 +296,36 @@ func (pg *PrivacyPage) Handle() {
...
-				Title("Warning: Danger alert!").
-				Body("Are you sure you want to allow spending from unmixed accounts?").
-				NegativeButton(values.String(values.StrCancel), func() {
+			textModal := modal.NewTextInputModal(pg.Load).
+				SetTextWithTemplate().
+				Hint("").
+				PositiveButtonStyle(pg.Load.Theme.Color.Danger, pg.Load.Theme.Color.InvText).
+				PositiveButton("Confirm", func(textInput string, tim *modal.TextInputModal) bool {
+					if textInput != "I understand the risks" {
+						pg.Toast.NotifyError("Please enter correct passphrase", 10)
```
Comment (Sirmorrison) [inline ui/page/wallets/privacy_page.go]:
```suggestion
						tim.SetError("confirmation text is incorrect")
```

File: ui/modal/info_modal_layouts.go [hunk 12]
```diff
@@ -121,3 +123,30 @@ func backupInfo(th *decredmaterial.Theme) []layout.Widget {
 		},
 	}
 }
+
+func allowUnspendUnmixedAcct(th *decredmaterial.Theme) []layout.Widget {
+	return []layout.Widget{
+		func(gtx C) D {
+			return layout.Flex{}.Layout(gtx,
+				layout.Rigid(func(gtx C) D {
+					ic := decredmaterial.NewIcon(decredmaterial.MustIcon(widget.NewIcon(icons.ContentReport)))
```
Comment (Sirmorrison) [inline ui/modal/info_modal_layouts.go]:
```suggestion
					ic := decredmaterial.NewIcon(decredmaterial.MustIcon(widget.NewIcon(icons.ActionInfo)))
```

File: ui/modal/info_modal_layouts.go [hunk 13]
```diff
@@ -121,3 +123,30 @@ func backupInfo(th *decredmaterial.Theme) []layout.Widget {
...
+					ic.Color = th.Color.Gray1
+					return ic.Layout(gtx, unit.Dp(18))
```
Comment (Sirmorrison) [inline ui/modal/info_modal_layouts.go]:
```suggestion
					return layout.Inset{Top: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
					  return ic.Layout(gtx, unit.Dp(18))
					})
```

File: ui/modal/text_input_modal.go [hunk 14]
```diff
@@ -87,9 +90,14 @@ func (tm *TextInputModal) SetCancelable(min bool) *TextInputModal {
 	return tm
 }
 
+func (tm *TextInputModal) SetTextWithTemplate() *TextInputModal {
+	tm.textCustomTemplate = allowUnspendUnmixedAcct(tm.Theme)
+	return tm
+}
```
Comment (Sirmorrison) [inline ui/modal/text_input_modal.go]: This section here needs to be dynamic just in case we need to use another template.

```suggestion
func (tm *TextInputModal) SetTextWithTemplate(template string) *TextInputModal {
	switch template {
	case AllowUnmixedSpendingTemplate:
		tm.textCustomTemplate = allowUnspendUnmixedAcct(tm.Theme)
	}

	return tm
}
}
```

File: ui/page/wallets/privacy_page.go [hunk 15]
```diff
@@ -295,25 +296,36 @@ func (pg *PrivacyPage) Handle() {
 
 	if pg.allowUnspendUnmixedAcct.Changed() {
 		if pg.allowUnspendUnmixedAcct.IsChecked() {
-			modal.NewInfoModal(pg.Load).
-				Title("Warning: Danger alert!").
-				Body("Are you sure you want to allow spending from unmixed accounts?").
-				NegativeButton(values.String(values.StrCancel), func() {
+			textModal := modal.NewTextInputModal(pg.Load).
+				SetTextWithTemplate().
```
Comment (Sirmorrison) [inline ui/page/wallets/privacy_page.go]:
```suggestion
				SetTextWithTemplate(modal.AllowUnmixedSpendingTemplate).
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Comment by dreacot** [comment]
why isn't the confirm button blue?

also noticed that the input accepts white spaces, white spaces should be trimmed

**Comment by Sirmorrison** [comment]
> why isn't the confirm button blue?
> 
> also noticed that the input accepts white spaces, white spaces should be trimmed

image

its the mockup design. Its a danger zone.

**Comment by AdimekweEbuka** [comment]
> why isn't the confirm button blue?
> 
> also noticed that the input accepts white spaces, white spaces should be trimmed

Looking through, majority of the editors accept whitespace input, another issue could be opened to tackle this.

**Comment by dreacot** [comment]
> > why isn't the confirm button blue?
> > also noticed that the input accepts white spaces, white spaces should be trimmed
> 
> Looking through, majority of the editors accept whitespace input, another issue could be opened to tackle this.

alright noted, we can do that

File: ui/modal/info_modal_layouts.go [hunk 16]
```diff
@@ -121,3 +123,32 @@ func backupInfo(th *decredmaterial.Theme) []layout.Widget {
...
+				layout.Rigid(func(gtx C) D {
+					ic := decredmaterial.NewIcon(decredmaterial.MustIcon(widget.NewIcon(icons.ActionInfo)))
+					ic.Color = th.Color.Gray1
+					return layout.Inset{Top: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
+						return ic.Layout(gtx, unit.Dp(18))
+					})
+				}),
+				layout.Rigid(func(gtx C) D {
+					text := th.Body1("Spendings from unmixed accounts could potentially be traced back to you")
+					text.Color = th.Color.Gray1
```
Comment (Sirmorrison) [inline ui/modal/info_modal_layouts.go]:
```suggestion
					text.Color = th.Color.GrayText1
```

File: ui/modal/info_modal_layouts.go [hunk 17]
```diff
@@ -121,3 +123,32 @@ func backupInfo(th *decredmaterial.Theme) []layout.Widget {
...
+					return layout.Inset{Left: values.MarginPadding5}.Layout(gtx, text.Layout)
+				}),
+			)
+		},
+
+		func(gtx C) D {
+			text := `<span style="text-color: gray">
+					Please type "<span style="text-color: black"> <span style="font-weight: bold">I understand the risks</span> </span>
+					" to allow spending from unmixed accounts.
+			</span>`
```
Comment (Sirmorrison) [inline ui/modal/info_modal_layouts.go]:
```suggestion
			text := `<span style="text-color: grayText1">
					Please type "<span style="font-weight: bold">I understand the risks</span>
					" to allow spending from unmixed accounts.
			</span>`
```

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Review by Sirmorrison — APPROVED** [review]
